service/src/fs_service.rs

```diff
@@ -131,6 +132,54 @@ pub trait FsService: Send + Sync {
         Ok(())
     }
 
+    /// Waits for the VFS to be fully initialized
+    fn wait_vfs_initialized(
+        &self,
+        vfs: &Arc<Vfs>,
+        max_wait_ms: u64,
+        interval_ms: u64,
+    ) -> Result<()> {
+        let max_wait = Duration::from_millis(max_wait_ms);
+        let interval = Duration::from_millis(interval_ms);
+        let start_time = Instant::now();
+        let mut last_log_time = Instant::now();
+        let log_interval = Duration::from_secs(1);
+
+        if vfs.initialized() {
+            info!("VFS already initialized.");
+            return Ok(());
+        }
+
+        info!("Waiting for VFS to be initialized...");
+        loop {
```

Comment:
Hi @Zephyrcf, maybe call `save_vfs_stat` after `initialized`, instead of waiting in a loop.